### PR TITLE
Generate lcov file for tests runs with coverage

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

**Review by tinder-maxwellelliott — APPROVED** [review]

**Comment by tinder-maxwellelliott** [comment]
Thanks for the assist on this. Learned a good deal reading this file

File: apple/testing/default_runner/ios_test_runner.template.sh [hunk 1]
```diff
@@ -146,10 +158,62 @@ else
...
+fi
+
+xcrun llvm-cov \
+  export \
+  -format lcov \
+  -instr-profile "$profdata" \
+  -path-equivalence="$ROOT","$PWD" \
+  "$test_binary" \
+  @"$COVERAGE_MANIFEST" \
+  > "$COVERAGE_OUTPUT_FILE" \
```
Comment (tinder-maxwellelliott) [inline apple/testing/default_runner/ios_test_runner.template.sh:209]: Do you have docs on these variables?

Comment (keith) [reply apple/testing/default_runner/ios_test_runner.template.sh:209]: don't think there is any, some are in here https://github.com/bazelbuild/bazel/blob/master/tools/test/collect_coverage.sh but I just ran `env; exit 1` in this script to see what was available. it does seem to treat these special (we just throw things in the undeclared outputs)

File: apple/testing/default_runner/ios_test_runner.template.sh [hunk 2]
```diff
@@ -146,10 +158,62 @@ else
...
+# llvm-cov export doesn't print any warnings/errors when files that it's trying
+# to reference don't exist. Unfortunately these are absolute paths that won't
+# be valid across machines. When using llvm-cov show it prints a warning for
+# this so we can validate we don't hit this case.
+error_file=$(mktemp)
+llvm_cov_status=0
+xcrun llvm-cov \
+  show \
+  -instr-profile "$profdata" \
+  -path-equivalence="$ROOT","$PWD" \
```
Comment (keith) [inline apple/testing/default_runner/ios_test_runner.template.sh:189]: this is the biggest part I'm unsure about, this isn't right for folks using the remote cache, they need to use coverage-prefix-map and remap `.` to `$PWD`

File: apple/testing/default_runner/ios_test_runner.template.sh (hunk 1 above)
Comment (tinder-maxwellelliott) [reply apple/testing/default_runner/ios_test_runner.template.sh:209]: TIL just go into Bazel stuff and do `env; exit 1`

File: apple/testing/default_runner/ios_test_runner.template.sh (hunk 2 above)
Comment (keith) [reply apple/testing/default_runner/ios_test_runner.template.sh]: I'll leave this and we can fix if something becomes more clear

**Comment by keith** [comment]
I pushed macOS coverage support here, @tinder-maxwellelliott can you try it?

**Comment by tinder-maxwellelliott** [comment]
> I pushed macOS coverage support here, @tinder-maxwellelliott can you try it?

Will take a look. Thanks again Keith

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
@tinder-maxwellelliott Were you able to take a look?

**Pull request merged**